src/wp-includes/script-loader.php

```diff
@@ -2879,7 +2879,17 @@ function wp_get_inline_script_tag( $javascript, $attributes = array() ) {
 	 *
 	 * @see https://www.w3.org/TR/xhtml1/#h-4.8
 	 */
-	if ( ! $is_html5 ) {
+	if (
+		! $is_html5 &&
+		(
+			! isset( $attributes['type'] ) ||
+			'module' === $attributes['type'] ||
+			str_contains( $attributes['type'], 'javascript' ) ||
+			str_contains( $attributes['type'], 'ecmascript' ) ||
+			str_contains( $attributes['type'], 'jscript' ) ||
+			str_contains( $attributes['type'], 'livescript' )
```

Comment:
This will account for a range of possible legacy types beyond the current `text/javascript`, such as: `application/javascript`, `text/x-javascript`, `text/javascript1.5`, `application/javascript`, and so on.

For example, see this list from WPRocket: https://github.com/wp-media/wp-rocket/blob/8d510d7b160011ff175488e17d9d6d1254ed16f9/inc/Engine/Optimization/DelayJS/HTML.php#L43-L59

See also a query on Bard: https://g.co/bard/share/60c27cafcc8f

Reply:
Is `type="module"` relevant here? I guess it may not be mutually exclusive, but in which world would someone not use HTML5 but JS modules? 😆

Reply:
Yes, it is relevant. A theme may not "support" HTML5 but a plugin may still add modules to the page.